msdk-io-mgf/src/main/java/io/github/msdk/io/mgf/MgfMsSpectrum.java

```diff
@@ -0,0 +1,108 @@
+package io.github.msdk.io.mgf;
+
+/*
+ * (C) Copyright 2015-2018 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+
+import com.google.common.collect.Range;
+import io.github.msdk.datamodel.MsSpectrum;
+import io.github.msdk.datamodel.MsSpectrumType;
+import io.github.msdk.util.tolerances.MzTolerance;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class MgfMsSpectrum implements MsSpectrum {
+  private MsSpectrumType type;
+  private double[] mz;
+  private float[] intensive;
+  private String title;
+  private int precursorCharge;
+  private double precursorMass;
+  private int size;
+
+  public MgfMsSpectrum(double[] mz, float[] intensive, int size, String title, int precursorCharge,
+      double precursorMass, MsSpectrumType type) {
+    this.title = title;
+    this.mz = mz;
+    this.intensive = intensive;
+    this.precursorMass = precursorMass;
+    this.precursorCharge = precursorCharge;
+    this.size = size;
+    this.type = type;
+  }
+
+  @Nonnull
+  @Override
+  public MsSpectrumType getSpectrumType() {
+    return type;
+  }
+
+  public String getTitle() {
+    return title;
+  }
+
+  public double getPrecursorMass() {
+    return precursorMass;
+  }
+
+  public int getPrecursorCharge() {
+    return precursorCharge;
+  }
+
+  @Nonnull
+  @Override
+  public Integer getNumberOfDataPoints() {
+    return size;
+  }
+
+  @Nonnull
+  @Override
+  public double[] getMzValues() {
+    return mz;
+  }
+
+  @Nonnull
+  @Override
+  public double[] getMzValues(double[] array) {
+    return new double[0];
```

Comment:
Are you sure this is doing the right thing?

Reply:
Oh, well it is a mistake I forgot about.
Actually what was it supposed to do? Accept array and return array?

Reply:
The purpose is to reuse arrays when possible. Take a look at this existing abstract class:
https://github.com/msdk/msdk/blob/master/msdk-datamodel/src/main/java/io/github/msdk/datamodel/AbstractMsSpectrum.java

Actually, your MgfMsSpectrum could extend the AbstractMsSpectrum, which would save you some coding.